vra/resource_deployment.go

```diff
@@ -152,22 +158,45 @@ func resourceDeploymentCreate(d *schema.ResourceData, m interface{}) error {
 	if err != nil {
 		switch err.(type) {
 		case *deployments.CheckDeploymentNameUsingGETNotFound:
-			log.Printf("deployment '%v' doesn't exist already and hence can be created", deploymentName)
+			log.Printf("Deployment '%v' doesn't exist already and hence can be created", deploymentName)
 		}
 	} else {
 		return fmt.Errorf("a deployment with name '%v' exists already. Try with a differnet name", deploymentName)
 	}
 
 	// If catalog_item_id is provided, request deployment with the catalog item
 	if catalogItemId != "" {
-		log.Printf("requesting vra_deployment '%s' from catalog item", d.Get("name"))
+		log.Printf("Requesting vra_deployment '%s' from catalog item", d.Get("name"))
+		catalogItemVersion := ""
+		if v, ok := d.GetOk("catalog_item_version"); ok {
+			catalogItemVersion = v.(string)
+		}
+
 		catalogItemRequest := models.CatalogItemRequest{
 			DeploymentName: deploymentName,
 			ProjectID:      projectId,
+			Version:        catalogItemVersion,
 		}
 
 		if v, ok := d.GetOk("inputs"); ok {
-			catalogItemRequest.Inputs = v
+			inputsSchemaMap, err := getCatalogItemSchema(apiClient, catalogItemId, catalogItemVersion)
+			if err != nil {
+				return err
+			}
+
+			log.Printf("Inputs Schema: %v", inputsSchemaMap)
+
+			inputTypesMap, err := getInputTypesMapFromCatalotItemSchema(inputsSchemaMap)
+			if err != nil {
+				return err
+			}
+
+			log.Printf("InputTypesMap: %v", inputTypesMap)
+
+			catalogItemRequest.Inputs, err = getInputsByType(v.(map[string]interface{}), inputTypesMap)
+			if err != nil {
+				return err
+			}
 		} else {
```

Comment:
Do we need this else here?

Reply:
Yes. Even if there are no Inputs, the services expect an empty map.